### ISSUE TITLE
3.1.100 + Publish + NoPIA adds COM lib to deps json file resulting in crash at runtime

### ISSUE THREAD

**Issue by 0xd4d** [dotnet/sdk#3993]
If you use .NET Core 3.1.100 SDK to publish a simple app with a COMReference and try to run it, it crashes at runtime because the deps json file has a reference to the COM DLL file.

Possibly related to this PR https://github.com/dotnet/sdk/pull/3539

```pwsh
# build it with .NET Core 3.1 SDK
msbuild -v:m -m -restore -t:Publish -p:Configuration=Release -p:TargetFramework=netcoreapp3.1 -p:RuntimeIdentifier=win-x64 -p:SelfContained=True
# works
.\bin\Release\netcoreapp3.1\win-x64\ConsoleApp19.exe
# crashes, see below
.\bin\Release\netcoreapp3.1\win-x64\publish\ConsoleApp19.exe
```

```
Error:
  An assembly specified in the application dependencies manifest (ConsoleApp19.deps.json) was not found:
    package: 'Interop.IWshRuntimeLibrary', version: '1.0.0.0'
    path: 'Interop.IWshRuntimeLibrary.dll'
```

This DLL file shouldn't even be in the deps json file and wasn't included by .NET Core SDK 3.0.

```csproj
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>netcoreapp3.1;netcoreapp3.0</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup>
    <COMReference Include="IWshRuntimeLibrary">
      <Guid>{F935DC20-1CF0-11D0-ADB9-00C04FD58A0B}</Guid>
      <VersionMajor>1</VersionMajor>
      <VersionMinor>0</VersionMinor>
      <Lcid>0</Lcid>
      <WrapperTool>tlbimp</WrapperTool>
      <Isolated>False</Isolated>
      <EmbedInteropTypes>True</EmbedInteropTypes>
    </COMReference>
  </ItemGroup>

</Project>
```

```C#
using System;

namespace ConsoleApp19
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
        }
    }
}
```

**Comment by 0xd4d** [comment]
Dupe of https://github.com/dotnet/sdk/issues/3981

**Comment by nguerrera** [comment]
Fixed by #4038
